SharkCage/CageNetwork/NetworkManager.cpp

```diff
@@ -5,76 +5,97 @@
 #include "NetworkManager.h"
 #include "MsgService.h"
 
+#include <chrono>
 
-DLLEXPORT NetworkManager::NetworkManager(ExecutableType type) : io_service(), socket(io_service), acceptor(io_service, tcp::endpoint(tcp::v4(), 1338))
-{
-	switch (type)
-	{
-	case ExecutableType::UI:
-		InitUi();
-		break;
-	case ExecutableType::SERVICE:
-		InitService();
-		break;
-	case ExecutableType::MANAGER:
-		InitManager();
-		break;
-	default:
-		break;
-	}
-}
+DLLEXPORT NetworkManager::NetworkManager(ContextType receive_for_type)
+{	
+	int receive_port = -1;
 
-DLLEXPORT std::wstring NetworkManager::Receive()
-{
-	rec_buf.clear();
-	rec_buf.resize(1024);
-	try
-	{
-		size_t len = socket.receive(asio::buffer(rec_buf));
-		rec_buf.resize(len);
-	}
-	catch (std::system_error e)
+	receive_port = GetPort(receive_for_type);
```

Comment:
Maybe we should get over the return if the function succeeded and over a passed variable into the function the port? So that we have not mixed an error case and a return value?

Reply:
will refactor the return value into an `optional`